js/src/@types/global.d.ts

```diff
@@ -1,3 +1,8 @@
+declare type Writable<T> = { -readonly [P in keyof T]: T[P] };
+declare type DeepWritable<T> = { -readonly [P in keyof T]: DeepWritable<T[P]> };
+
+declare type DeepReadonly<T> = { -readonly [P in keyof T]: DeepReadonly<T[P]> };
```

Comment:
I think it should be this, actually.

```suggestion
declare type DeepReadonly<T> = { readonly [P in keyof T]: DeepReadonly<T[P]> };
```